Generated CLI does not compile when using arrays as query parameters

## Problem statement
When generating a CLI from the enclosed swagger spec the code produced does not compile.
The code that is generated tries to assign a value of type `*[]int64` to a variable of type `[]int64`.
```bash
➜ go build ./gen/cmd/cli                                 
# github.com/kevinbarbour/scratch/gen/cli
gen/cli/get_pet_operation.go:99:9: cannot use &idsFlagValues (type *[]int64) as type []int64 in assignment
```

Is seems that the problem comes from the retrieveflag template assuming that any Nullable parameter should be assigned as a pointer which is not true for these slices.

## Swagger specification
```yaml
swagger: "2.0"
info:
  title: "sample"
  version: "1"
paths:
  /pet:
    get:
      parameters:
      - in: "query"
        name: "ids"
        type: "array"
        items:
          type: "integer"
      responses:
        200:
          description: "good"
```

## Steps to reproduce
Generate a CLI from the above swagger spec and attempt to run or compile it.

## Environment
go version: go1.17.8
OS:  macOS 12.2.1

--- a/generator/templates/cli/retrieveflag.gotmpl
+++ b/generator/templates/cli/retrieveflag.gotmpl
@@ -38,7 +38,7 @@
             return err, false
         }
         {{- /* reciever by convention is m for CLI */}}
-        m.{{ pascalize .Name }} = {{- if .IsNullable }}&{{- end }}{{ $flagValueVar }}
+        m.{{ pascalize .Name }} = {{ $flagValueVar }}
     {{- else if or (eq .GoType "[]strfmt.DateTime") (eq .GoType "[]strfmt.ObjectId") (eq .GoType "[]strfmt.UUID") }} {{/*Get flag value as string, then parse it*/}}
         {{ template "flagnamevar" . }}
         {{ $flagValueVar }}Str, err := cmd.Flags().GetStringSlice({{ $flagNameVar }})

